front/app/components/SignUpIn/SignUp/index.tsx

```diff
@@ -113,281 +112,268 @@ export interface InputProps {
+      )
+    ) {
+      await completeRegistration(registrationData);
+    }
+  };
+
+  // this makes sure that if registration is completed,
+  // but we're not in a modal, handleFlowCompleted is
+  // still called even without closing the Success window
+  useEffect(() => {
+    if (
+      !isNilOrError(authUser) &&
+      !!authUser.attributes.registration_completed_at &&
+      !metaData.inModal
+    ) {
+      handleFlowCompleted();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [authUser, metaData]);
+
+  const onResize = (_width, height) => {
+    setHeaderHeight(`${Math.round(height) + 2}px`);
+  };
+
+  const handleSelectAuthProvider = (
+    selectedAuthProvider: AuthProvider,
+    setHrefFromModule?: () => void
+  ) => {
+    if (selectedAuthProvider === 'email') {
+      setEmailSignUpSelected(true);
+    } else {
+      handleOnSSOClick(selectedAuthProvider, metaData, setHrefFromModule);
+    }
+  };
+
+  const handleStepError = (errorMessage?: string) => {
+    errorMessage
+      ? setError(errorMessage)
+      : setError(formatMessage(messages.somethingWentWrongText));
+  };
+
+  const handleFlowCompleted = () => {
+    trackEventByName(tracks.signUpFlowCompleted);
+    onSignUpCompleted();
+  };
+
+  const handleOnOutletData = (
+    configuration: TSignUpStepConfigurationObject
+  ) => {
+    setConfiguration((oldConfiguration) => ({
+      ...oldConfiguration,
+      [configuration.key]: configuration,
+    }));
+  };
+
+  const handleOnOutletDataLoaded = (step: TSignUpStep, loaded: boolean) => {
+    setDataLoadedPerOutlet((oldDataLoadedPerOutlet) => ({
+      ...oldDataLoadedPerOutlet,
+      [step]: loaded,
+    }));
+  };
+
+  const handleGoBack = () => {
+    setEmailSignUpSelected(false);
+  };
+
+  // emit event whenever activeStep changes
+  useEffect(() => signUpActiveStepChange(activeStep), [activeStep]);
+
+  useEffect(() => {
+    if (metaData?.error) {
+      setError(formatMessage(messages.somethingWentWrongText));
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [metaData?.error]);
+
+  const helpText = activeStepConfiguration?.helperText?.(
+    !isNilOrError(tenant) ? tenant.data : undefined
+  );
+
+  const stepDescription = activeStepConfiguration?.stepDescriptionMessage
+    ? formatMessage(activeStepConfiguration.stepDescriptionMessage)
+    : '';
+
+  return (
+    <Container id="e2e-sign-up-container" className={className ?? ''}>
+      {activeStep !== 'success' && (
+        <Header
+          inModal={metaData.inModal}
+          onResize={onResize}
+          activeStepNumber={activeStepNumber}
+          totalStepsCount={totalStepsCount}
+          error={error}
+          stepName={stepDescription}
+        />
+      )}
+
+      <StyledModalContentContainer
+        inModal={!!metaData.inModal}
+        windowHeight={`${windowHeight}px`}
+        headerHeight={headerHeight}
+        ref={modalContentRef}
+      >
+        {error ? (
```

Comment:
This is just personal preference and not related to any of your changes, but for big ternary conditions like this we can save some indentation in the future by doing it something like this:

```
if (error) { 
    return <Error text={error} animate={false} marginBottom="30px" />
}

return (<>
    {helpText && (
     <SignUpHelperText [...]
```